File: Angular2_AspDotNet/API/Angular2_AspDotNet.Web/src/app/pages/user/user.model.ts
export class UserViewModel {

    public UserID: number;
    public UserName: string;
    public FirstName: string;
    public LastName: string;
    public EmailID: string;
    public OrganizationID: number;
    public RoleID: number;
    public SecurityQuestion1: number;
    public SecurityQuestion2: number;
    public SecurityAnswer1: string;
    public SecurityAnswer2: string;
    public OverRideOrganizationId: boolean = false;
    public RoleList: number[];

    constructor() {
    }
}

export class RoleViewModel {
    public RoleId: number;
    public RoleName: string;
    public IsAdmin: boolean;

    constructor() {
    }
}

export class SecurityQuestionModel {
    constructor() {
    }

    public SecurityQuestionID: number;
    public Question: string;
}

export class OrgaizationViewModel {
    constructor() {
    }

    public OrganizationID: number;
    public OrganizationName: string;
}
export class UserViewListModel {
    constructor() {
    }
    public UserID: number;
    public UserName: string;
    public FirstName: string;
    public LastName: string;
    public EmailID: string;
    public OrganizationID: number;
    public RoleID: number;
    public SecurityQuestion1: number;
    public SecurityQuestion2: number;
    public SecurityAnswer1: string;
    public SecurityAnswer2: string;
    public OrganizationName: string;
    public RoleName: string;
    public SQ1: string;
    public SQ2: string;
    public RowID: number;
    public InActive: boolean;
    public InActiveText: string;
}

export class UserViewList {
    constructor() {
    }
    public UserViewListModel: UserViewListModel[];
    public TotalRecords: number;
}

export class UserResetPassword {
    public OldPassword: string;
    public Password: string;
    public ConfirmPassword: string;
    constructor() {
    }
}

File: Angular2_AspDotNet/API/Angular2_AspDotNet.Models/UserModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;
using System.Threading.Tasks;

namespace Angular2_AspDotNet.Models
{
    [DataContract]
    public class UserModel
    {
        [DataMember]
        public int id { get { return UserID; } }
        [DataMember]
        public string name
        {
            get { return FirstName + " " + LastName; }
            set { }
        }
        public int UserID { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
    }

    public class UserViewModel
    {


        public int UserID { get; set; }
        public string UserName { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string EmailID { get; set; }
        public Nullable<int> OrganizationID { get; set; }
        public Nullable<int> RoleID { get; set; }
        public Nullable<int> SecurityQuestion1 { get; set; }
        public Nullable<int> SecurityQuestion2 { get; set; }
        public string SecurityAnswer1 { get; set; }
        public string SecurityAnswer2 { get; set; }
        public string Password { get; set; }
        public Nullable<System.DateTime> Addedon { get; set; }
        public Nullable<int> AddedBy { get; set; }
        public Nullable<bool> IsNewUser { get; set; }
        public List<int> RoleList { get; set; } = new List<int>();
        public bool OverRideOrganizationId { get; set; } = false;

    }

    public class UserViewListModel
    {
        public int UserID { get; set; }
        public string UserName { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string EmailID { get; set; }
        public Nullable<int> OrganizationID { get; set; }
        public Nullable<int> RoleID { get; set; }
        public Nullable<int> SecurityQuestion1 { get; set; }
        public Nullable<int> SecurityQuestion2 { get; set; }
        public string SecurityAnswer1 { get; set; }
        public string SecurityAnswer2 { get; set; }
        public string OrganizationName { get; set; }
        public string RoleName { get; set; }
        public string SQ1 { get; set; }
        public string SQ2 { get; set; }
        public Nullable<long> RowID { get; set; }
        public Nullable<bool> InActive { get; set; }
        public string InActiveText
        {
            get
            {
                if (InActive == true)
                    return "Yes";
                else
                    return
                        "No";
            }
        }
    }

    public class UserViewList
    {
        public List<UserViewListModel> UserViewListModel { get; set; }
        public int TotalRecords { get; set; }
    }

    public class UserMenuModel
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string AddedDate { get; set; }
        public int UserId { get; set; }
    }

    public class UserResetPassword
    {
        public string OldPassword { get; set; }
        public string Password { get; set; }
        public string ConfirmPassword { get; set; }

    }

    public class GetUserWithRoleViewResult
    {
        public string UserName { get; set; }
        public int UserID { get; set; }
        public Nullable<int> RoleId { get; set; }
    }
}
